command/pipe.go

```diff
@@ -0,0 +1,360 @@
+		},
+		&cli.StringFlag{
+			Name:  "cache-control",
+			Usage: "set cache control for target: defines cache control header for object, e.g. pipe --cache-control 'public, max-age=345600'",
+		},
+		&cli.StringFlag{
+			Name:  "expires",
+			Usage: "set expires for target (uses RFC3339 format): defines expires header for object, e.g. pipe  --expires '2024-10-01T20:30:00Z'",
+		},
+		&cli.BoolFlag{
+			Name:  "force-glacier-transfer",
+			Usage: "force transfer of glacier objects whether they are restored or not",
+		},
+		&cli.BoolFlag{
+			Name:  "ignore-glacier-warnings",
+			Usage: "turns off glacier warnings: ignore errors encountered during copying, downloading and moving glacier objects",
+		},
+		&cli.StringFlag{
+			Name:  "destination-region",
+			Usage: "set the region of destination bucket: the region of the destination bucket will be automatically discovered if --destination-region is not specified",
+		},
+		&cli.BoolFlag{
+			Name:  "raw",
+			Usage: "disable the wildcard operations, useful with filenames that contains glob characters",
+		},
+		&cli.StringFlag{
+			Name:  "content-type",
+			Usage: "set content type for target: defines content type header for object, e.g. --content-type text/plain",
+		},
+		&cli.StringFlag{
+			Name:  "content-encoding",
+			Usage: "set content encoding for target: defines content encoding header for object, e.g. --content-encoding gzip",
+		},
+		&cli.IntFlag{
+			Name:        "no-such-upload-retry-count",
+			Usage:       "number of times that a request will be retried on NoSuchUpload error; you should not use this unless you really know what you're doing",
+			DefaultText: "0",
+			Hidden:      true,
+		},
+		&cli.BoolFlag{
+			Name:    "no-clobber",
+			Aliases: []string{"n"},
+			Usage:   "do not overwrite destination if already exists",
+		},
+	}
+	return pipeFlags
+}
+
+func NewPipeCommand() *cli.Command {
+	cmd := &cli.Command{
+		Name:               "pipe",
+		HelpName:           "pipe",
+		Usage:              "upload to remote from stdin",
+		Flags:              NewPipeCommandFlags(),
+		CustomHelpTemplate: pipeHelpTemplate,
+		Before: func(c *cli.Context) error {
+			err := validatePipeCommand(c)
+			if err != nil {
+				printError(commandFromContext(c), c.Command.Name, err)
+			}
+			return err
+		},
+		Action: func(c *cli.Context) (err error) {
+			defer stat.Collect(c.Command.FullName(), &err)()
+
+			pipe, err := NewPipe(c, false)
+			if err != nil {
+				return err
+			}
+			return pipe.Run(c.Context)
+		},
+	}
+
+	cmd.BashComplete = getBashCompleteFn(cmd, false, false)
+	return cmd
+}
+
+// Pipe holds pipe operation flags and states.
+type Pipe struct {
+	dst         *url.URL
+	op          string
+	fullCommand string
+
+	deleteSource bool
+
+	// flags
+	noClobber             bool
+	storageClass          storage.StorageClass
+	encryptionMethod      string
+	encryptionKeyID       string
+	acl                   string
+	forceGlacierTransfer  bool
+	ignoreGlacierWarnings bool
```

Comment:
In `cp`, Glacier options are only valid for source objects;
https://github.com/peak/s5cmd/blob/74d21bb4f022edb0bfb69f412c4baa69153ba8c2/command/cp.go#L410-L417
https://github.com/peak/s5cmd/blob/74d21bb4f022edb0bfb69f412c4baa69153ba8c2/command/cp.go#LL358-L364


I think those can be omitted here as there aren't any objects, just stdin.

Reply:
Yes, we don't need them. I have removed.
https://github.com/peak/s5cmd/pull/587/commits/3b6d0a8dfb260b215a6601a6ee4f1269b6c35c54